src/ExactOnline.Client.Sdk/Helpers/ExactOnlineQuery.cs

```diff
@@ -394,6 +394,7 @@ private static string ToODataParameter(object value)
 					: type == typeof(Guid) ? $"guid'{value}'"
 					: type == typeof(DateTime) ? $"datetime'{value:o}'"
 					: type == typeof(bool) ? value.ToString().ToLower()
+					: type == typeof(long) ? $"{value}L"

```

Comment:
This is already checked 2 lines lower... maybe the order with the "IsEnum" check needs to be changed?